lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -350,29 +361,30 @@ def main():
 
         # If VPC configuration is desired
         if vpc_subnet_ids or vpc_security_group_ids:
-            if len(vpc_subnet_ids) < 1:
-                module.fail_json(msg='At least 1 subnet is required')
-
-            if len(vpc_security_group_ids) < 1:
-                module.fail_json(msg='At least 1 security group is required')
+            try:
+                assert len(vpc_subnet_ids) > 0
+                assert len(vpc_security_group_ids) > 0
+            except (AssertionError, TypeError):
+                module.fail_json(msg='vpc connectivity requires at least one security group and one subnet')
 
             if 'VpcConfig' in current_config:
                 # Compare VPC config with current config
-                current_vpc_subnet_ids = current_config['VpcConfig']['SubnetIds']
-                current_vpc_security_group_ids = current_config['VpcConfig']['SecurityGroupIds']
+                current_subnet_ids = current_config['VpcConfig']['SubnetIds']
+                current_security_group_ids = current_config['VpcConfig']['SecurityGroupIds']
 
-                subnet_net_id_changed = sorted(vpc_subnet_ids) != sorted(current_vpc_subnet_ids)
-                vpc_security_group_ids_changed = sorted(vpc_security_group_ids) != sorted(current_vpc_security_group_ids)
+                subnets_changed = sorted(vpc_subnet_ids) != sorted(current_subnet_ids)
```

Comment:
I don't see the value in changing these var names - can you change them back to what they were to preserve authorship?